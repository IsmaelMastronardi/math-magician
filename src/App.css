.pageLinks {
  background-color: lightcoral;
  height: 60px;
  width: 300px;
  position: fixed;
  right: 0;
}

.navList {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  list-style: none;
  gap: 20px;
  padding: 0;
  margin: 0;
  height: 100%;
}

a:visited,
a:hover,
a:link,
a:active {
  color: black;
  text-decoration: none;
  font-size: 1.5rem;
}