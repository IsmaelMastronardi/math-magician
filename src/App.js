import { Route, Routes, Link } from 'react-router-dom';
import Calculator from './components/Calculator';
import ComputerQuotes from './components/quotes';
import HomePage from './components/Home';
import './App.css';
import './styles/homePage.css';
import './styles/calculator.css';
import './styles/navbar.css';

function App() {
  return (
    <>
      <nav className="pageLinks flexRow">
        <h1 className="pageTitle">Math Magicians</h1>
        <ul className="navList flexRow">
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/calculator">Calculator</Link>
          </li>
          <li>
            <Link to="/quotes">Quotes</Link>
          </li>
        </ul>
      </nav>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quotes" element={<ComputerQuotes />} />
      </Routes>
    </>
  );
}

export default App;
