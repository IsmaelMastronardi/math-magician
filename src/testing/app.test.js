/* eslint-disable import/no-extraneous-dependencies */
import { MemoryRouter } from 'react-router';
import { render } from '@testing-library/react';
import App from '../App';

describe('Renders App', () => {
  test('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
