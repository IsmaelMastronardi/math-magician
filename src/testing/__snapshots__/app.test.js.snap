// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App renders correctly 1`] = `
<div>
  <nav
    class="pageLinks flexRow"
  >
    <h1
      class="pageTitle"
    >
      Math Magicians
    </h1>
    <ul
      class="navList flexRow"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="homeDiv"
  >
    <h2>
      Welcome to my Page!
    </h2>
    <p
      class="homeText"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p
      class="homeText"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>
</div>
`;
